src/Repository/UserRepository.php

```diff
@@ -3,27 +3,69 @@
 namespace App\Repository;
 
 use App\Entity\User\User;
+use Exception;
+use App\Exceptions\EntityNotFoundException;
 
-class UserRepository
+
+class UserRepository extends Repository
 {
-    private DBConnexion $dbConnection;
+    protected $table = 'user';
 
-    public function __construct(DBConnexion $dbConnection)
+    public function __construct(\App\Repository\DBConnexion $dbConnection)
     {
-        $this->dbConnection = $dbConnection;
+        try{
+            parent::__construct($dbConnection, $this->table, User::class);
+
+            if(!$this->entity){
+                throw new EntityNotFoundException($this->entity);
+            }
+        }catch(Exception $exception){
+           $e = new EntityNotFoundException($this->entity);
+           echo $e->getClass() . ': '. $exception->getMessage();
+        }
     }
 
-    /**
-     * @return User[]
-     */
-    public function findAll(): array
+    //CREATE NEW USER BY PASSING AN OBJECT
+    public function createUser(User $user)
     {
-        $result = $this->dbConnection->execute('SELECT*FROM user');
-        $users = [];
-        foreach ($result as $item){
-            $users[] = new User($item['firstname'], $item['name'],$item['email'], $item['status'], $item['bio'],$item['password']);
+        try{
+            $insertInto = $this->dbConnection->getPDO()->prepare('INSERT INTO user (firstname, name, email, status, bio, password) VALUES (?,?,?,?,?,?)');
+            return $insertInto ->execute(array($user->getFirstname(), $user->getName(), $user->getEmail(), $user->getStatus(), $user->getBio(),$user->getPassword()));
         }
-        return $users;
+        catch(\PDOException $exception){
+            echo 'The following error has occured :' . $exception->getMessage() . ' at ligne ' . $exception->getLine() . ' in the following file: ' . $exception->getFile();
```

Comment:
when you have an erreur you should throw an exception. If you want to log information for dev, you should use loffer interface from psr3.